### DISCUSSION TITLE
[Feature] Text based search

### DISCUSSION THREAD

**Discussion by ayykamp** [immich-app/immich#7264] [Feature Request]
### The feature

Hey, I really like the new search :) One improvement I could see would be a sort of "power user" search, like github does it.
![image](https://github.com/immich-app/immich/assets/32194363/d7ee72be-d297-498a-a37a-83f8617e7b73)
So being able to provide all the filter with tags in the text search like c:camera or iso:200. This would make it much easier to use with a keyboard only. Github also gives you nice autocompletes so that also might be worth looking. I know you can use "m:" already to search all of the metadata which is very usefult but a more fine  grained approach would be nice. Thanks

### Platform

- [ ] Server
- [x] Web
- [x] Mobile

**Comment by alextran1502** [comment]
Yeah we have plan for this following the search filter

**Reply by lprimak** [reply]
Please consider "smart approx location search" functionality as well as described in #19388

**Comment by stefanos-manganaris** [comment]
new to immich and love it -- please consider including the PATH of an asset in the searchable metadata -- looking for a way to include/exclude assets based on the library they live in, in a setting with internal and multiple external libraries / import paths

**Comment by dgtal1** [comment]
I'm new to immich and I definitely think the Smart Search function needs to be improved as currently it gives me completely unpredictable results. I do not understand the relation between my query and the results I get.

Introducing github-style tags may address this, but I proposed a smarter (although probably more complex to implement) solution here: #8903

**Comment by lprimak** [comment]
> I have photos that I took with the Family in Coney Island (Brooklyn, NY)
When I search for in the location pulldown for Brooklyn NY, I can find the photos.
However, when I enter "coney island" in the search bar, the photo is not found.

> Search bar has "Context: my search term" Apparently the context doesn't include location?
I would say that it should be included.

> I've checked the location on the map in the photo, and it definitely includes "coney island" there.
Also, Apple Photos can find the same photo by searching "coney island"

> Flickr, Google Photos and Apple Photos have this functionality.

See #19388 for complete details

**Comment by mbuzina** [comment]
Ok, I will comment my feature idea here. If the machine learning algorithm also gets the information about people in the picture, it should be able to infer things like "tom on a swing" to Tom being the person on the swing and not Tim with Tom standing next to it. That way you can search fluently (searching for "tom on a swing" instead of "p:tom on a swing")

**Comment by Mrlimache** [comment]
This is somewhat what I have been searching for 
Most of the popular photo apps have a single search that includes basic search such as location and dates as well as context DOG, CAT, etc. I see #8903 suggests something similar. When I search "Monkey Japan 2026" I would expect it to Identify location and time as a filter then searches 'Monkey' as context. I know it's is achievable via the drop-down filters, but in practice that's slow and detracts from the user experience IMO. 

Other than that, I'm really enjoying the app and am supporting it, amazed by the progress. Its already a must use app for myself and my users but the last "wife approval factor" is the search as it's a bit of a pain point for the less tech savvy (or lazy) users.
